src/routes/controllers/projects/getAllProjects.js

```diff
@@ -0,0 +1,141 @@
+                        FROM "Project" 
+                        WHERE id IN (SELECT project_id 
+                                    FROM "ProjectSkill"
+                                    WHERE skill = ANY ($1)
+                                    GROUP BY project_id
+                                    HAVING COUNT(DISTINCT skill) = array_length($1, 1))
+                                AND id IN (SELECT project_id 
+                                    FROM "ProjectTag"
+                                    WHERE tag = ANY ($2)
+                                    GROUP BY project_id
+                                    HAVING COUNT(DISTINCT tag) = array_length($1, 1))`,
+                        [req.query.skill, req.query.tag]
+                    );
+                } else if (req.query.skill) {
+
+                        projects = await pool.query(
+                            `SELECT id,name,description,creator,status,created_at 
+                            FROM "Project" 
+                            WHERE id IN (SELECT project_id 
+                                        FROM "ProjectSkill"
+                                        WHERE skill = ANY ($1)
+                                        GROUP BY project_id
+                                        HAVING COUNT(DISTINCT skill) = array_length($1, 1))`,
+                            [req.query.skill]
+                        );
+                } else if (req.query.tag) {
+                    projects = await pool.query(
+                        `SELECT id,name,description,creator,status,created_at 
+                        FROM "Project" 
+                        WHERE id IN (SELECT project_id 
+                                    FROM "ProjectTag"
+                                    WHERE tag = ANY ($1)
+                                    GROUP BY project_id
+                                    HAVING COUNT(DISTINCT tag) = array_length($1, 1))`,
+                        [req.query.tag]
+                    );
+                }
+            } 
+        }
+
+        let promises = [];
+
+        // get skills of projects
+        promises.push(
+            ...projects.rows.map(async (row) => {
+                const skills = await pool.query({
+                    text: 'SELECT skill FROM "ProjectSkill" WHERE project_id = $1',
+                    values: [row.id],
+                    rowMode: 'array',
+                });
+                row.skills = skills.rows.flat();
+                return row;
+            })
+        );
+
+        // get tags of projects
+        promises.push(
+            ...projects.rows.map(async (row) => {
+                const tags = await pool.query({
+                    text: 'SELECT tag FROM "ProjectTag" WHERE project_id = $1',
+                    values: [row.id],
+                    rowMode: 'array',
+                });
+                row.tags = tags.rows.flat();
+                return row;
```

Comment:
It seems like this could have been refactored into a helper function since you're repeating code in another controller as well. Don't trip over it, though.